### PR TITLE
basic sample: separated into 2 samples (xml vs. programmatic) versus 1

### PR DESCRIPTION
**Pull request by lanimall**
To keep things simple and consistent with the way it's done for the clustered sample, i refactored the Basic sample into 2 samples (BasicXML and BasicProgrammatic)...

### CODE REVIEW

File: basic/pom.xml [hunk 1]
```diff
@@ -1,63 +1,93 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-  <modelVersion>4.0.0</modelVersion>
-  <groupId>org.ehcache.sample</groupId>
-  <artifactId>basic</artifactId>
-  <version>1.0.0-SNAPSHOT</version>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (ljacomet) [inline basic/pom.xml]: Changes here look like a simple move from two spaces to 4 spaces - please remove them from the PR.

File: clustered/src/main/resources/ehcache.xml [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
 
-  <ehcache:service>
-    <terracotta:cluster>
-      <terracotta:connection url="terracotta://localhost:9510/clustered"/>
-      <terracotta:server-side-config auto-create="true">
-        <terracotta:default-resource from="default-resource"/>
-      </terracotta:server-side-config>
-    </terracotta:cluster>
- </ehcache:service>
+    <ehcache:service>
```
Comment (ljacomet) [inline clustered/src/main/resources/ehcache.xml]: Here again, please stick with the existing formatting: two spaces, not 4.

File: jsr107/src/main/java/org/ehcache/sample/BaseJCacheTester.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Created by fabien.sanglier on 10/8/16.
+ */
+public abstract class BaseJCacheTester {
+    private static final Logger LOGGER = getLogger(CreateExtendedJCache.class);
+
+    protected <K, V> boolean simpleIterationCacheTest(Cache<K, V> myJCache, int numberOfIteration, int numberOfObjectPerIteration, int sleepTimeMillisBetweenIterations, KeyValueGenerator<K, V> keyValueGenerator) throws InterruptedException {
```
Comment (ljacomet) [inline jsr107/src/main/java/org/ehcache/sample/BaseJCacheTester.java]: I would rename this test method - I thought it was about iterating entries of the cache - which it is not (and that's good 😉 )

**Comment by lanimall** [comment]
ok thanks for all the comments. I think I fixed all the format issues and other issues you caught. Let me know how this PR looks now. thanks.

File: clustered/pom.xml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (ljacomet) [inline clustered/pom.xml]: Changes here are irrelevant - can you drop them from the PR?

File: clustered/src/main/java/org/ehcache/sample/ClusteredXML.java [hunk 5]
```diff
@@ -20,9 +20,9 @@ public static void main(String[] args) {
     Configuration xmlConfig = new XmlConfiguration(myUrl);
     try (CacheManager cacheManager = newCacheManager(xmlConfig)) {
       cacheManager.init();
-      
+
```
Comment (ljacomet) [inline clustered/src/main/java/org/ehcache/sample/ClusteredXML.java]: This file contains only formatting changes - please remove from the PR

File: clustered/src/main/resources/ehcache.xml [hunk 6]
```diff
@@ -12,7 +14,7 @@
         <terracotta:default-resource from="default-resource"/>
       </terracotta:server-side-config>
     </terracotta:cluster>
- </ehcache:service>
+  </ehcache:service>
```
Comment (ljacomet) [inline clustered/src/main/resources/ehcache.xml]: Same here - only formatting changes - please remove from the PR

**Comment by ljacomet** [comment]
Looks good on the newly added code - but I request you drop the unrelated formatting changes. Make sure your IDE does not reformat sections you do not edit.

**Comment by lanimall** [comment]
Ok thanks for your patience... Tried to fix all the formatting-only changes.
Let me know if anything else is off now.

**Pull request merged**